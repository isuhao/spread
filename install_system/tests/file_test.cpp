#include "common.cpp"

int main()
{
  SetupTest ss("_file");

  {
    Setup s("Empty run");
  }
  {
    Setup s("Add files, no source");
    s.addFile("test1/hello", hello);
    s.addFile("test1/world", world);
  }
  {
    Setup s("Add files, with source");
    s.addFile("test2/hello", hello);
    s.addFile("test2/world", world);
    s.index("data/hello.dat");
    s.index("data/world.dat");
  }
  {
    Setup s("Same file repeated");
    s.index("data/hello.dat");
    s.addFile("test3/hello", hello);
    s.addFile("test3/hello2", hello);
    s.addFile("test3/dir/hello3", hello);
  }
  {
    Setup s("Overwriting existing file that matches");
    s.index("data/hello.dat");
    s.addFile("test3/hello", hello);
  }
  {
    Setup s("Overwriting ditto, but without having file in cache");
    s.addFile("test3/hello", hello);
  }
  {
    Setup s("Adding a custom dir");
    s.index("data/hello.dat");
    s.index("data/world.dat");
    s.addDir(dir1, "test4");
  }
  {
    Setup s("Overwriting without cache");
    s.addDir(dir1, "test4");
  }
  {
    Setup s("Adding dir twice + subdir");
    s.index("data/hello.dat");
    s.index("data/world.dat");
    s.addDir(dir1, "test5");
    s.addDir(dir1, "test5");
    s.addDir(dir1, "test5/sub");
  }
  {
    Setup s("Adding dir as hash, no source");
    s.addDir(Hash("NOFILE"), "test6");
  }
  {
    Setup s("Adding dir as hash, with source but missing files");
    Hash dirH = s.index("data/dir1.dat");
    s.addDir(dirH, "test6");
  }
  {
    Setup s("Adding dir as hash, should work now");
    Hash dirH = s.index("data/dir1.dat");
    s.index("data/hello.dat");
    s.index("data/world.dat");
    s.addDir(dirH, "test6");
  }

  return 0;
}
