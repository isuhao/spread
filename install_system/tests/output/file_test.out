
START: Empty run
RESULT:
  SUC: Success 0/0

START: Add files, no source
LOG: ERROR: No source for target LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF _file/test1/hello
GOT ERROR: No source for target LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF _file/test1/hello
RESULT:
  SUC: Success 0/0
    ---
    ERR: No source for target LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF _file/test1/hello 0/0

START: Add files, with source
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/world.dat
RESULT:
  SUC: Success 0/0

START: Same file repeated
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
RESULT:
  SUC: Success 0/0

START: Overwriting existing file that matches
RESULT:
  SUC: Success 0/0

START: Overwriting ditto, but without having file in cache
RESULT:
  SUC: Success 0/0

START: Adding a custom dir
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/world.dat
RESULT:
  SUC: Success 0/0

START: Overwriting without cache
RESULT:
  SUC: Success 0/0

START: Adding dir twice + subdir
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/world.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/world.dat
RESULT:
  SUC: Success 0/0

START: Adding dir as hash, no source
LOG: Hash NOFILE does NOT match any archive rule, assuming dirHash
LOG: ERROR: No source for target NOFILE 
GOT ERROR: No source for target NOFILE 
RESULT:
  SUC: Success 0/0
    ---
    ERR: No source for target NOFILE  0/0

START: Adding dir as hash, with source but missing files
LOG: Hash 1GxiMyo3usMorjw5J4Gtn-Avw6ZrL2muqEcbyP-G5iyX does NOT match any archive rule, assuming dirHash
LOG: Loaded DIR dirHash=1GxiMyo3usMorjw5J4Gtn-Avw6ZrL2muqEcbyP-G5iyX
LOG: ERROR: No source for target LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF _file/test6/abc
GOT ERROR: No source for target LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF _file/test6/abc
RESULT:
  SUC: Success 0/0
    ---
    ERR: No source for target LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF _file/test6/abc 0/0

START: Adding dir as hash, should work now
LOG: Hash 1GxiMyo3usMorjw5J4Gtn-Avw6ZrL2muqEcbyP-G5iyX does NOT match any archive rule, assuming dirHash
LOG: Loaded DIR dirHash=1GxiMyo3usMorjw5J4Gtn-Avw6ZrL2muqEcbyP-G5iyX
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/hello.dat
LOG: STATUS: Copying file /home/mortennk/koding/spread/install_system/tests/data/world.dat
RESULT:
  SUC: Success 0/0

Directory: _file
00                                                 cache/
00                                                 cache/1G/
1GxiMyo3usMorjw5J4Gtn-Avw6ZrL2muqEcbyP-G5iyX       cache/1G/xiMyo3usMorjw5J4Gtn-Avw6ZrL2muqEcbyP-G5iyX
00                                                 test2/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test2/hello
SG6kYiTRu0-2gPNPfJrZao8k7Ii-c-qOWmxlJg6cuKcF       test2/world
00                                                 test3/
00                                                 test3/dir/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test3/dir/hello3
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test3/hello
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test3/hello2
00                                                 test4/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test4/abc
00                                                 test4/def/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test4/def/ghi
00                                                 test4/jkl/
SG6kYiTRu0-2gPNPfJrZao8k7Ii-c-qOWmxlJg6cuKcF       test4/jkl/mno
00                                                 test5/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test5/abc
00                                                 test5/def/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test5/def/ghi
00                                                 test5/jkl/
SG6kYiTRu0-2gPNPfJrZao8k7Ii-c-qOWmxlJg6cuKcF       test5/jkl/mno
00                                                 test5/sub/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test5/sub/abc
00                                                 test5/sub/def/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test5/sub/def/ghi
00                                                 test5/sub/jkl/
SG6kYiTRu0-2gPNPfJrZao8k7Ii-c-qOWmxlJg6cuKcF       test5/sub/jkl/mno
00                                                 test6/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test6/abc
00                                                 test6/def/
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF       test6/def/ghi
00                                                 test6/jkl/
SG6kYiTRu0-2gPNPfJrZao8k7Ii-c-qOWmxlJg6cuKcF       test6/jkl/mno
Total 35 elements
Hash: 3Jx-q1qotYno4ezRuWcK1rNKwjIX2YWsyMffNvVyBhxnBw
